tests/integration/test_plugins.py

```diff
@@ -34,10 +39,37 @@ async def test_jenkins_plugins_config(
     res = unit_web_client.client.requester.get_url(unit_web_client.web)
     page_content = str(res.content, encoding="utf-8")
 
-    assert all(plugin in page_content for plugin in plugins_meta.remove), page_content
+    assert all(plugin in page_content for plugin in REMOVED_PLUGINS), page_content
     assert "The following plugins have been removed by the system administrator:" in page_content
     assert (
         "To allow the plugins, please include them in the plugins configuration of the charm."
         in page_content
     )
-    assert all(unit_web_client.client.has_plugin(plugin) for plugin in plugins_meta.config)
+    assert all(unit_web_client.client.has_plugin(plugin) for plugin in ALLOWED_PLUGINS)
+
+
+@pytest.mark.usefixtures("app_k8s_agent_related")
+async def test_git_plugin_k8s_agent(ops_test: OpsTest, unit_web_client: UnitWebClient):
+    """
+    arrange: given a jenkins charm with git plugin installed.
+    act: when a job is dispatched with a git workflow.
+    assert: job completes successfully.
+    """
+    await install_plugins(
+        ops_test, unit_web_client.unit, unit_web_client.client, INSTALLED_PLUGINS
+    )
+
+    job_name = "git-plugin-test-k8s"
+    unit_web_client.client.create_job(job_name, gen_git_test_job_xml("k8s"))
+    # check that git plugin git repository validation works on Jenkins server
+    check_url_res = unit_web_client.client.requester.post_url(
+        f"{unit_web_client.client.baseurl}/job/{job_name}/descriptorByName/"
+        "hudson.plugins.git.UserRemoteConfig/checkUrl",
+        data={
+            "value": "https://github.com/canonical/jenkins-k8s-operator",
+            "credentialsId": "",
+        },
+    )
+    assert (
```

Comment:
I'm not very familiar with Jenkins, so just to get the idea: You submit the job using the post_url with the data and then you know that it was completed if there is a "<div/>" in the content response, is that right?

Reply:
Yup, it returns any error messages in HTML, in this case, there were no errors and hence an empty `<div />` is returned.